beacon-chain/rpc/service.go

```diff
@@ -252,6 +264,44 @@ func (s *Service) Start() {
 			}
 		}
 	}()
+	if featureconfig.Get().EnableSlasherConnection {
+		s.startSlasherClient()
+	}
+}
+
+func (s *Service) startSlasherClient() {
+	var dialOpt grpc.DialOption
+	if s.slasherCert != "" {
+		creds, err := credentials.NewClientTLSFromFile(s.slasherCert, "")
+		if err != nil {
+			log.Errorf("Could not get valid credentials: %v", err)
+			s.slasherCredentialError = err
+		}
+		dialOpt = grpc.WithTransportCredentials(creds)
+	} else {
+		dialOpt = grpc.WithInsecure()
+		log.Warn("You are using an insecure gRPC connection! Please provide a certificate and key to use a secure connection.")
+	}
+	slasherOpts := []grpc.DialOption{
+		dialOpt,
+		grpc.WithStatsHandler(&ocgrpc.ClientHandler{}),
+		grpc.WithStreamInterceptor(middleware.ChainStreamClient(
+			grpc_opentracing.StreamClientInterceptor(),
+			grpc_prometheus.StreamClientInterceptor,
+		)),
+		grpc.WithUnaryInterceptor(middleware.ChainUnaryClient(
+			grpc_opentracing.UnaryClientInterceptor(),
+			grpc_prometheus.UnaryClientInterceptor,
+		)),
+	}
+	conn, err := grpc.DialContext(s.ctx, s.slasherProvider, slasherOpts...)
+	if err != nil {
+		log.Errorf("Could not dial endpoint: %s, %v", s.slasherProvider, err)
+		return
+	}
+	log.Info("Successfully started gRPC connection")
```

Comment:
This log is too vague, gRPC connection to where?

Reply:
updated